code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -166,6 +166,11 @@
 		to_chat(X, "<span class='warning'>We can only shape on weeds. We must find some resin before we start building!</span>")
 		return fail_activate()
 
+	var/obj/machinery/door/poddoor/timed_late/containment/shutter = locate(/obj/machinery/door/poddoor/timed_late/containment) in range(8, owner.loc)
```

Comment:
This is disgusting
Locate and range are both awful rework this to be sane

Reply:
I was literally told by fucking splinter to do this, I told him it was bad